client/src/components/Navbar/Navbar.tsx

```diff
@@ -1,34 +1,44 @@
 import Logo from '../../assets/Logo.png';
-import {
-  NavigationMenu,
-  NavigationMenuItem,
-  NavigationMenuList,
-  NavigationMenuTrigger,
-} from '@/components/ui/navigation-menu.jsx';
 import { Input } from '@/components/ui/input.tsx';
 import { Button } from '../ui/button.tsx';
 import ModeToggle from '../ModeToggle.tsx';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { SearchInput } from '@/types/index.ts';
+import {
+  useSearchParams,
+  createSearchParams,
+  useNavigate,
+  NavLink,
+} from 'react-router-dom';
 
 export default function Navbar() {
+  const { register, handleSubmit } = useForm<SearchInput>();
+  const [searchParams, setSearchParams] = useSearchParams({ query: '' });
+  const navigate = useNavigate();
+  const submitHandler: SubmitHandler<SearchInput> = (formData) => {
+    setSearchParams({ query: formData.query });
+    const url = createSearchParams({ query: formData.query }).toString();
+    navigate(`/search?${url}`);
+  };
```

Comment:
Put a check if the query string is empty then the page won't navigate to `/search`

A minor suggestion : instead of triggering the search event on click we can use a debouncer to improve the UX for the search feat.